### ISSUE TITLE
Syntax Error after cleanup at token '='

### ISSUE THREAD

**Issue by this-yash** [SAP/abap-cleaner#226]
Hi,

I've been facing an issue if I use FOR expression with GROUP BY clause. If I comment it, the ABAP Cleaner works fine. 
Screenshots are attached below for both the sample code and the issue. Please, share if it can be fixable from my side else check if it is fixable from your side.
Also, thanks for the awesome tool. It's a game changer!

![image](https://github.com/SAP/abap-cleaner/assets/64050540/010f4497-83f0-4805-95c4-33bef6b5d98c)
![image](https://github.com/SAP/abap-cleaner/assets/64050540/cfa554ce-69f3-4252-b68d-584d4fd43724)

**Comment by jmgrassau** [comment]
Hi Yash, 

thanks a lot for reporting this issue! This seems to be a bug in the cleanup rule "Shorten VALUE statements", which mistakes the group key for a table row. A minimal example to trigger the error would be this:

```ABAP
itab = VALUE #( FOR GROUPS OF <fs> IN itab GROUP BY 
                ( plant = <fs>-plant )
                ( plant = <fs>-plant
                  l1    = <fs>-l1 ) ).
```

Here, the cleanup rule erroneously identifies ```plant = <fs>-plant``` as a common assignment and therefore extracts it from both inner parentheses …

```ABAP
itab = VALUE #( FOR GROUPS OF <fs> IN itab GROUP BY 
                plant = <fs>-plant
                ( )
                ( l1    = <fs>-l1 ) ).
```

… which is a syntax error. Will fix this asap!

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Yash, 

this bug is now fixed for the next release.

Kind regards,
Jörg-Michael

**Comment by this-yash** [comment]
Hi,

Thank you very much.  Will wait for the next release.

**Comment by jmgrassau** [comment]
Hi Yash, 

thanks again for this bug report – this error should now be fixed with version 1.13.0, which was just released!

Kind regards,
Jörg-Michael

**Comment by ConjuringCoffee** [comment]
I love learning about syntax possibilities from ABAP Cleaner bug reports 😄
